### DISCUSSION TITLE
set_width for console

### DISCUSSION THREAD

**Discussion by aroberge** [Textualize/rich#1262] [Ideas] [answered]
Using either an IPython console or a Jupyter notebook/lab, I often find that the default width is too wide. I defined a simple `set_width` function as follows:
```python
def set_width(width):
    console_instance._width = width
```
which I use in interactive sessions.

This works perfectly for my use case. However, I know that accessing private variables (hidden behind a read-only property) is definitely not good practice, but I could not find any other way to update the parameters of an active console.  Am I missing a more acceptable way to achieve the same result?

**Comment by willmcgugan** [comment, marked as answer]
The next version of Rich will have a property setter for width and height. In the meantime, I forgive you for accessing a private variable! It won't break.

**Reply by aroberge** [reply]
Thanks using `width` instead of `_width` works ... in most cases. However, I found that `__rich_repr__` does not seem to respect the console's width (regardless of whether I use `width` or `_width`. Is is supposed to respect the console width, thus indicating that I have done something wrong?

**Reply by aroberge** [reply]
Perhaps I should clarify: the `__rich_repr__` content consists of long strings. 

If I "physically" reduce the width of the terminal, they do not wrap **properly** at the boundary either.  Here's an example of this:
![image](https://user-images.githubusercontent.com/629698/124967133-7274b000-dffa-11eb-8766-f087e456ae0e.png)

**Reply by willmcgugan** [reply]
The Rich reprs will format as you would code. It will expand on to multiple lines where it can to fit within the width. But if a single line contains a string that is larger than the width there is nowhere to insert a new line that doesn't break the syntax -- and it leaves the wrapping to the terminal.

**Reply by aroberge** [reply]
Thanks, it makes sense.  I am abusing `__rich_repr__` to provide translatable help in a convenient way.

When I have time, I will write a blob post about how I "use and abuse" Rich in my project.
